sale_brand/wizard/sale_make_invoice_advance.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleAdvancePaymentInv(models.TransientModel):
+    _inherit = "sale.advance.payment.inv"
+
+    @api.multi
+    def _create_invoice(self, order, so_line, amount):
+        invoice = super()._create_invoice(order, so_line, amount)
+        invoice.brand_id = order.brand_id
+        invoice._onchange_partner_id()
```

Comment:
Would it make sense to have a more focused `_on_change_brand_id()?` 
If yes this can be done in another PR.

Reply:
`_on_change_brand_id` exists in #14

Reply:
BTW, I assume there is no prepare method in that wizard?

Reply:
> BTW, I assume there is no prepare method in that wizard?

Unfortunately yes

Reply:
> `_on_change_brand_id` exists in #14

I see what you mean. When this PR is merged I will make the changes in #14